docs/proposals/scheduling/20240201-enbale-reservation-preempt.md

```diff
@@ -0,0 +1,186 @@
+suppose the pods bound to the preempted reservations still need to be evicted. Hence, users need to extend the implementation 
+to set reservation priority.
+
+#### Story2
+Only allow reservations can preempt reservations when failed to schedule. Pod is not allowed to preempt reservations. 
+
+#### Story3
+Typically, the reservations purchased by tenants are homogeneous, and it will benefit for tenants’ training workloads if
+the reservations are scheduled in the same topology unit. Therefore, we aim to do reservation preemption in batches, too. 
+
+### Implementation Details
+If reservation failed to schedule after PreFilter and Filter, and the reservation can trigger preemption, scheduler
+can trigger preemption in PostFilter. During the preemption, the reservation can preempt the lower priority reservations.
+
+But by default, the koord-scheduler sets the priority of reserved pod (constructed by Reservation) to Int32Max to disable 
+preempting reservation even if the preemptor also is Reservation. So we introduce new label 
+`scheduling.koordinator.sh/reservation-priority` to set priority according to the needs. If koord-scheduler notices that 
+Reservation.Labels has labels, use that value as priority, otherwise keep the default behavior.
+`scheduling.koordinator.sh/reservation-priority` should be defined as Metadata.Labels of Reservation. The higher the value, 
+the higher the priority. 
+
+Typically, users use preemptionPolicy to declare whether reservation can trigger preemption or not. If the preemptionPolicy 
+is PreemptLowerPriority, the reservation can trigger preemption. However, in de-scheduler users may already set 
+Reservation.Spec.Template.Spec.PreemptionPolicy with `PreemptLowerPriority`, so we need to compatible with the existing 
+reservation pods. We suppose to introduce label `scheduling.koordinator.sh/can-preempt` to judge whether reservation can 
+preempt or not.
+
+If the reservation uses `pod-group.scheduling.sigs.k8s.io`, then the batch preemption will be supported by job-level 
+preemption.
+
+#### Preemption Rule
+The reservation preemption still follows the existing Filter/PostFilter procedure and can be combined with job-level
+preemption mechanism. 
+
+The rules of implementing the preemption strategy of reservation are as followed:
+1. Only `scheduling.koordinator.sh/can-preempt=true` reservation can trigger preemption, and only lower priority 
+   reservations can be preempted.
+
+2. Reservations can only preempt reservations.
+
+3. If only part of Reservation can be assigned successfully in preemption dry-run process, the preemption will not
+   really happen.
+
+4. If reservation is preempted, the bound pods should also be evicted.
+
+5. Users are expected to implement the soft-eviction mechanism. If the soft-eviction mechanism is not implemented, users 
+   should delete the evicted reservation and pods once the reservation is preempted.
```

Comment:
> users should delete the evicted reservation and pods once the reservation is preempted.

scheduler should delete the reservation and pods .... ?

Reply:
done